packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -55,8 +54,72 @@ const AccountSignup = ({
         setPWInput(new_password);
     };
 
+    let eventQueue = [];
+
+    const queueEvent = event => {
+        eventQueue.push(event);
+        localStorage.setItem('pending_events', JSON.stringify(eventQueue));
+    };
+
+    const trackEvent = event => {
+        if (window.rudderanalytics) {
+            console.log('here');
+            window.rudderanalytics?.track(event.name, event.properties);
+        } else {
+            console.log('else');
+            queueEvent(event);
+        }
+    };
+
+    const loadPendingEvents = () => {
+        const storedEvents = localStorage.getItem('pending_events');
+        console.log('pending events', storedEvents);
+        if (storedEvents) {
+            console.log('storedEvents');
+            eventQueue = JSON.parse(storedEvents);
+        }
+    };
+
+    const initializeAnalytics = async () => {
+        try {
+            await AnalyticsInitializer();
+            console.log('analytics loaded');
+
+            if (eventQueue.length > 0) {
+                console.log('eventQueue.length');
+                eventQueue.forEach(event => {
+                    window.rudderanalytics.track(event.name, event.properties);
+                });
+
+                eventQueue = [];
+                localStorage.removeItem('pending_events');
+            }
+        } catch (error) {
+            console.error('Analytics failed to load', error);
+        }
+    };
+
+    console.log('before calling analytics');
+    loadPendingEvents();
```

Comment:
```suggestion
```
this also can be moved to the utility function, and it also can be handled out of the box from utility function, so the end user doenst need to worry about this.

Reply:
In the end you will export a trackEvent function which sends the pedingEvents if there's any.